clients/drcachesim/analyzer.cpp

```diff
@@ -339,19 +382,27 @@ analyzer_tmpl_t<RecordType, ReaderType>::process_tasks(analyzer_worker_data_t *w
             return;
         }
         int shard_index = worker->stream->get_input_stream_ordinal();
-        if (shard_data.find(shard_index) == shard_data.end()) {
+        if (worker->shard_data.find(shard_index) == worker->shard_data.end()) {
             VPRINT(this, 1, "Worker %d starting on trace shard %d stream is %p\n",
                    worker->index, shard_index, worker->stream);
-            shard_data[shard_index].resize(num_tools_);
+            worker->shard_data[shard_index].resize(num_tools_);
             for (int i = 0; i < num_tools_; ++i) {
-                shard_data[shard_index][i] = tools_[i]->parallel_shard_init_stream(
-                    shard_index, user_worker_data[i], worker->stream);
+                worker->shard_data[shard_index][i] =
+                    tools_[i]->parallel_shard_init_stream(
+                        shard_index, user_worker_data[i], worker->stream);
             }
+            worker->cur_interval_index = 0;
```

Comment:
See the other comments: this is assuming a back-to-back scheduling which is not promised by the scheduler.

Reply:
Fixed it. Tracking analyzer_per_shard data inside the worker data.